sdk/core/src/main/java/com/klaviyo/coresdk/Klaviyo.kt

```diff
@@ -18,112 +18,127 @@ object Klaviyo {
 
     /**
      * Configure Klaviyo SDK with your account's public API Key and application context.
-     * Optionally specify additional behavior customization
-     *
-     * This must be called to initialize the SDK before using any other functionality
+     * This must be called to before using any other SDK functionality
      *
      * @param apiKey - Your Klaviyo account's public API Key
      * @param applicationContext
-     * @param networkTimeout
-     * @param networkFlushInterval
-     * @param networkFlushDepth
-     * @param networkFlushCheckInterval
-     * @param networkUseAnalyticsBatchQueue
      * @return
      */
-    fun configure(
+    fun initialize(
         apiKey: String,
-        applicationContext: Context,
-        networkTimeout: Int = KlaviyoConfig.NETWORK_TIMEOUT_DEFAULT,
-        networkFlushInterval: Int = KlaviyoConfig.NETWORK_FLUSH_INTERVAL_DEFAULT,
-        networkFlushDepth: Int = KlaviyoConfig.NETWORK_FLUSH_DEPTH_DEFAULT,
-        networkFlushCheckInterval: Int = KlaviyoConfig.NETWORK_FLUSH_CHECK_INTERVAL,
-        networkUseAnalyticsBatchQueue: Boolean = KlaviyoConfig.NETWORK_USE_ANALYTICS_BATCH_QUEUE,
-    ) = apply {
+        applicationContext: Context
+    ) {
         KlaviyoConfig.Builder()
             .apiKey(apiKey)
             .applicationContext(applicationContext)
-            .networkTimeout(networkTimeout)
-            .networkFlushInterval(networkFlushInterval)
-            .networkFlushDepth(networkFlushDepth)
-            .networkFlushCheckInterval(networkFlushCheckInterval)
-            .networkUseAnalyticsBatchQueue(networkUseAnalyticsBatchQueue)
             .build()
+
+        // TODO should we guard all other APIs against being called before this?
+        // TODO initialize state from persistent store
+        // TODO initialize profile with new anon ID (if one was not found in store)
     }
 
     //region Fluent setters
 
     /**
-     * Assigns an email address to the current UserInfo
+     * Assigns an email address to the current internally tracked profile
      *
-     * UserInfo is saved to keep track of current profile details and
-     * used to autocomplete analytics requests with profile identifier when not specified
+     * The SDK keeps track of current profile details to
+     * build analytics requests with profile identifiers
      *
      * This should be called whenever the active user in your app changes
      * (e.g. after a fresh login)
      *
      * @param email Email address for active user
+     * @return
      */
-    fun setEmail(email: String) = apply {
+    fun setEmail(email: String): Klaviyo = apply {
+        // TODO validate email format?
```

Comment:
What would we do with email addresses we can't validate though?

Reply:
Throw I guess, Noah pointed out this might be too much clientside intelligence.

Reply:
Was thinking the same. I'm not sure if I'm just overlooking some use case that would be handy, but I feel like we're pretty agnostic here and will just pass up to the API whatever we have been given and let the `CustomerProcessor` within Klaviyo figure it out

Reply:
Thanks, I wanted to hear at least one other person's feedback but I figured I'd remove this comment.